### PR TITLE
Add eslint

### PR DESCRIPTION
**Pull request by ZuzaOsa**
Added eslint to check for issues in code for example space after ``//``.
- Rule about using prototypes was switched off beacuse Datastax code violates it. After switching the code to use classes the rule should be switched on.
- Added camelcase, no-var and spaced-comment rules.
- no-unused-vars rule was switched off for error catches and unused function parameters.
- While changing things to satisfy linter some geometry leftovers were removed in ``encoder.js``. Spaces were added in the comments. In ``integer.js`` some variables were switched from using snake_case to camelCase and ``var`` replaced with ``let`` (i have *several* questions about that code by the way).

### CODE REVIEW

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      "no-redeclare": "off",
+      "no-useless-escape": "off",
+      "no-unexpected-multiline": "off",
+      "no-cond-assign": "off",
+      "require-yield": "off",
+      // "camelcase": "error",
+      "spaced-comment": "error",
+    }
+  },
+];
```
Comment (adespawn) [inline eslint.config.mjs:27]: ;)

File: eslint.config.mjs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    rules: {
+      "no-undef": "off",
+      "no-unused-vars": "off",
+      "no-prototype-builtins": "off",
+      "no-redeclare": "off",
+      "no-useless-escape": "off",
+      "no-unexpected-multiline": "off",
+      "no-cond-assign": "off",
+      "require-yield": "off",
+      // "camelcase": "error",
```
Comment (adespawn) [inline eslint.config.mjs:23]: If we were to add lint, why disable all those errors?

File: test/integration/not-supported/client-execute-prepared-tests.js [hunk 3]
```diff
@@ -2289,23 +2289,23 @@ describe("Client @SERVER_API", function () {
...
+            // /  '1' !== '3'
+            // /
+            // /        + expected - actual
+            // /
+            // /        -1
+            // /        +3
+            // /
+            // /        at /home/runner/work/scylladb-javascript-driver/scylladb-javascript-driver/test/integration/short/client-execute-prepared-tests.js:2332:40
+            // /        at /home/runner/work/scylladb-javascript-driver/scylladb-javascript-driver/lib/promise-utils.js:147:44
+            // /        at process.processTicksAndRejections (node:internal/process/task_queues:77:11)
```
Comment (adespawn) [inline test/integration/not-supported/client-execute-prepared-tests.js:2308]: Breaking changes?

File: test/integration/not-supported/client-metadata-tests-long.js [hunk 4]
```diff
@@ -12,14 +12,14 @@ describe("Client", function () {
...
-    ///          -5
-    ///          +3
+    // /    Uncaught AssertionError [ERR_ASSERTION]: Expected values to be strictly equal:
+    // /
+    // /    '5' !== '3'
+    // /
+    // /          + expected - actual
+    // /
+    // /          -5
+    // /          +3
```
Comment (adespawn) [inline test/integration/not-supported/client-metadata-tests-long.js:22]: breaking changes?

**Review by adespawn — CHANGES_REQUESTED** [review]
Is it possible to add linter to ``pre-push`` command?

File: test/integration/not-supported/client-execute-prepared-tests.js (hunk 3 above)
Comment (ZuzaOsa) [reply test/integration/not-supported/client-execute-prepared-tests.js:2308]: Uno reverse - why are there ``///`` there?

File: eslint.config.mjs (hunk 2 above)
Comment (ZuzaOsa) [reply eslint.config.mjs:23]: Those errors are on in default linter options, but they are violated in Datastax code. I planned to add an issue for investigating and fixing them.

Comment (adespawn) [reply eslint.config.mjs:23]: In this case I would prefer to have those changes in this PR

File: lib/types/index.js [hunk 5]
```diff
@@ -481,7 +481,7 @@ FrameHeader.getProtocolVersion = function (buffer) {
  * @returns {FrameHeader}
  */
 FrameHeader.fromBuffer = function (buf, offset) {
-    let streamId = 0;
+    let streamId;
```
Comment (adespawn) [inline lib/types/index.js:484]: Is it even used?
If no, why doesn't it trigger eslint?

File: lib/types/integer.js [hunk 6]
```diff
@@ -1,6 +1,6 @@
 /** @module types */
 
-var utils = require("../utils");
+let utils = require("../utils");
```
Comment (adespawn) [inline lib/types/integer.js:3]: const imports

File: lib/types/integer.js
Comment (adespawn) [inline lib/types/integer.js:1]: That's a mess indeed

**Review by adespawn — COMMENTED** [review]
I have also several questions about removed eslint comments. Why they are no longer needed?

**Comment by ZuzaOsa** [comment]
> I have also several questions about removed eslint comments. Why they are no longer needed?

There were leftover comments from Datastax, they used different ruleset for eslint and unused disables were removed when running eslint with ``--fix``.

File: lib/execution-profile.js [hunk 7]
```diff
@@ -3,7 +3,6 @@
 const utils = require("./utils");
 const types = require("./types");
 const promiseUtils = require("./promise-utils");
-const { throwNotSupported } = require("./new-utils");
```
[the remark this answers is not in the document]
Comment (ZuzaOsa) [reply lib/execution-profile.js:6]: image
You don't? I do.

File: lib/types/index.js (hunk 5 above)
Comment (ZuzaOsa) [reply lib/types/index.js:484]: Yeah, it is used: 
image

Comment (adespawn) [reply lib/types/index.js:484]: I'm totally blind

File: lib/readers.js [hunk 8]
```diff
@@ -370,14 +370,14 @@ class FrameReader {
             meta.lastContinuousPage = !!(flags & resultFlag.lastContinuousPage);
         }
         if (flags & resultFlag.globalTablesSpec) {
-            meta.global_tables_spec = true;
+            meta.global_tables_spec = true; // eslint-disable-line camelcase
```
[the remark this answers is not in the document]
Comment (ZuzaOsa) [reply lib/readers.js:373]: I am not sure were it is used or if it is not exposed in the API (i could not figure it out).

**Review by adespawn — APPROVED** [review]

**Pull request merged**